### PR TITLE
fixed #4230 'Opening notifications in "Read" section still marks them read and removes them from the list'.

### LINKED ISSUE #4230: Opening notifications in "Read" section still marks them as read and removes them from the list.
**Summary:** 

On tapping a notification in read section of notifications, the app behaves as if in unread section and marks the notifications as read, removing them from the list.

**Steps to reproduce:** 
* Go to read the section in the notification.
* Tap on an item to view it.
* Get back to see notification removed from read section.

What did you expect the app to do, and what did you see instead?
Notification should not be removed.

**Device and Android version:** 
Nokia 6.1+ android 10 stock.
 
**Commons app version:** 
Latest master.

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by 4D17Y4**
**Description (required)**

Fixes #4230

What changes did you make and why?

Added a private variable which is true when in read section of the notifications.
Used it whenever required.

**Tests performed (required)**

Tested 3.0.0-notification on Nokia 6.1+ android 10.

### CODE REVIEW

**Comment by 4D17Y4** [comment]
Up for review @nicolas-raoul, @neslihanturan :)

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java [hunk 1]
```diff
@@ -64,11 +64,13 @@
     private NotificationWorkerFragment mNotificationWorkerFragment;
     private NotificatinAdapter adapter;
     private List<Notification> notificationList;
+    private boolean isRead;
     MenuItem notificationMenuItem;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        isRead=getIntent().getStringExtra("title").equals("read");
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:73]: Style: One space before and after `=`.

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java [hunk 2]
```diff
@@ -87,6 +89,9 @@ public boolean onSupportNavigateUp() {
 
     @SuppressLint("CheckResult")
     public void removeNotification(Notification notification) {
+        if(isRead) {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:92]: Style: One space after `if` like in the similar lines you added as part of this commit.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
It fixes the issue, thanks!
Minor detail, would you mind fixing the spacing?

**Comment by 4D17Y4** [comment]
Done.
Thanks.

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java [hunk 3]
```diff
@@ -64,11 +64,13 @@
     private NotificationWorkerFragment mNotificationWorkerFragment;
     private NotificatinAdapter adapter;
     private List<Notification> notificationList;
+    private boolean isRead;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:67]: Ah sorry l forgot one last thing: Could you please add javadoc for this?
A sentence explaining with words what a true value means.
I know other members have no javadoc, but our rule is to add javadoc to all new classes/members/methods. Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java (hunk 3 above)
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java]: Added javadoc.

File: app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java [hunk 4]
```diff
@@ -85,8 +90,21 @@ public boolean onSupportNavigateUp() {
...
+    /**
+     * If this is unread section of the notifications, removeNotification method
+     *  Marks the notification as read,
+     *  Removes the notification from unread,
+     *  Displays the Snackbar.
+     *
+     * Otherwise returns (read section).
+     *
+     * @param notification
+     */
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java:102]: Thanks! :-)

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**